src/test/utils/meta_tables/meta_table_test.cpp

```diff
@@ -138,4 +144,18 @@ TEST_P(MultiMetaTablesTest, SQLFeatures) {
   EXPECT_EQ(result.second->row_count(), 1);
 }
 
+TEST_F(MetaTableTest, SingleGeneration) {
+  auto mock_table = std::make_shared<MetaMockTable>();
+  _add_meta_table(mock_table);
+
+  EXPECT_EQ(mock_table->generate_calls(), 0);
+  SQLPipelineBuilder{"SELECT * FROM meta_mock"}.create_pipeline().get_result_table();
+  EXPECT_EQ(mock_table->generate_calls(), 1);
+  SQLPipelineBuilder{"DELETE FROM meta_mock WHERE mock='abc'"}.create_pipeline().get_result_table();
+  EXPECT_EQ(mock_table->generate_calls(), 2);
+  SQLPipelineBuilder{"INSERT INTO meta_mock VALUES('foo')"}.create_pipeline().get_result_table();
+  EXPECT_EQ(mock_table->generate_calls(), 3);
+  SQLPipelineBuilder{"UPDATE meta_mock SET mock='foo'"}.create_pipeline().get_result_table();
+  EXPECT_EQ(mock_table->generate_calls(), 4);
+}
```

Comment:
Could you (1) make sure that the LQP and PQP caches are enabled and (2) repeat calls to make sure that even with caching, the tables are regenerated?